FirebaseDatabaseUI/FUICollection.h

```diff
@@ -18,57 +18,109 @@
 
 // clang-format on
 
+@import Foundation;
+
 @class FUIArray;
+@protocol FUICollectionDelegate;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A protocol representing a collection of objects from Firebase Database.
+ */
+@protocol FUICollection <NSObject>
+
+@property (nonatomic, readonly, copy) NSArray<FIRDataSnapshot *> *items;
+
+@property (weak, nonatomic, nullable) id<FUICollectionDelegate> delegate;
+
+/**
+ * The number of objects in the collection.
+ */
+@property (nonatomic, readonly) NSUInteger count;
+
+/**
+ * The @c FIRDataSnapshot at the given index. May raise fatal errors
+ * if the index is out of bounds.
+ * @param index The index of a snapshot.
+ */
+- (FIRDataSnapshot *)snapshotAtIndex:(NSInteger)index;
+
+/**
+ * Calling this makes the array begin observing updates from its query.
+ * Before this call is made the array is inert and doesn't do anything.
+ */
+- (void)observeQuery;
+
+/**
+ * Cancels all active observations. The array may be reused after this
+ * is called by calling @c observeQuery again.
+ */
+- (void)invalidate;
+
+@end
 
 /**
  * A protocol to allow instances of FUIArray to raise events through a
  * delegate. Raises all Firebase events except FIRDataEventTypeValue.
  */
-@protocol FUIArrayDelegate<NSObject>
+@protocol FUICollectionDelegate<NSObject>
 
 @optional
 
 /**
+ * Called before any other events are sent.
+ */
+- (void)arrayDidBeginUpdates:(id<FUICollection>)collection;
```

Comment:
This is just to better mirror the UIKit aggregation methods of [`beginUpdates`](https://developer.apple.com/reference/uikit/uitableview/1614908-beginupdates) and [`endUpdates`](https://developer.apple.com/reference/uikit/uitableview/1614890-endupdates).